app/models/work_version.rb

```diff
@@ -64,7 +64,11 @@ class WorkTypeUpdateError < RuntimeError; end
 
     event :deposit_complete do
       transition depositing: :deposited
+    end
+
+    event :pid_assigned do
       transition reserving_purl: :purl_reserved
+      transition %i[first_draft depositing pending_approval purl_reserved] => same
```

Comment:
Originally, I had expected only needing to include `first_draft` and `depositing`, but I guess there's an `sdr.objects.created`/`h2.druid_assigned` message in other situations too?  I added `pending_approval` and `purl_reserved` based (respectively iirc) on running unit tests and doing manual testing on stage.

Reply:
I'm a bit confused as to how first_draft. pending_approval or purl_reserved would ever make this transition.  Can you explain more?

Reply:
iirc, i added all three because either the unit tests tried to make that that transition, or because the app tried to make that transition when i tested on -stage on friday.  will try to reproduce and give a bit more detail as to what triggered this transition on those states...

Reply:
(i was also surprised, esp on the `purl_reserved` one)

Reply:
I think the unit test should be adjusted so that they are in the correct state rather than change the code to try and accomodate objects that aren't correct.

Reply:
fixed the test issue, but deploying to -qa to see if i can reproduce the transition issue i stumbled on when testing on -stage...

Reply:
full disclosure: i tested just now with this commit:  0329901ce6cb9ea3b71e3883da03fcd8ef3b4695

but that just adds some debug logging to the changes in this branch, in an attempt to figure out why a PURL res isn't turned into a real deposit after a user selects a type (and edits metadata and hits the "Deposit" button).

https://github.com/sul-dlss/happy-heron/compare/b5d6de56c686c8e8fea3f834a6fdd2d86cab5b30...0329901ce6cb9ea3b71e3883da03fcd8ef3b4695

Reply:
paired with @jcoyne this morning, and after walking through the code, we realized that `pid_assigned!` was getting called twice on a `WorkVersion` in `purl_reserved`, because `DepositJob` was being run twice for PURL reservations (because both the `reserve_purl` and `begin_deposit` events were being invoked for PURL reservation, and both of those events have an `after_transition` hook that invokes `WorkObserver#after_begin_deposit`).

this should be fixed in the third commit on this PR, and now `depositing` and `reserving_purl` should be the only "from" states for the `pid_assigned` event/transition.